### ISSUE TITLE
Enable innawoods butchering

### ISSUE THREAD

**Issue by kevingranade** [CleverRaven/Cataclysm-DDA#24878]
**Is your feature request related to a problem? Please describe.**  
As a wilderness-based survivor, I need to be able to fully butcher animals.

**Describe the solution you'd like**  
The simplest solution seems to be:
Allow skinning animals with just a knife (this could also target meat and other products, but need to be able to get skin with just a knife).
Add a "skin tarp" item/recipe to make a tarp from some amount of skins.
Add the "FLAT_FURN" flag to the tarp furniture.
Side note, a rack is already readily constructable.

**Describe alternatives you've considered**  
Fast butcher should still yield other products, in fact the "hard" products like bones should have full yield, and skinning the corpse (yield would be limited by skill of course), seems like a reasonable default as it's incredibly useful.
Other table-like constructions you can make without materials harvested from buildings would be nice, but the tarp seems the most thematic.

**Additional context**  
Add any other context (such as mock-ups, proof of concepts or screenshots) about the feature request here.

**Comment by FulcrumA** [comment]
As mentioned in related ticket, I'd simply make full butchering benefit from additional, dedicated devices/items which in turn would increase minimal yield and decrease time of butchering - but wouldn't be necessary to perform the action itself.

**Comment by Odieman1** [comment]
I've been researching this topic quite in depth and have found a few things.

First, I'd say that the size of the animal has a lot to do with how much you benefit from using a full set of utensils and tables.  I rabbit can be field butchered with much higher efficiency than a moose for example.  [Here's a video of a man completely cleaning out a rabbit in 3 minutes, I'm sure in another 10 he could de-bone it as well. (Graphic)](https://www.youtube.com/watch?v=GeFr_HuAR1M)

Large animals like Moose take exponentially longer.  [Here's a video of a guy taking about 15 minutes just to partially quarter a moose.](https://www.youtube.com/watch?v=wqUkawJhbLw).  A moose weighs about 800-1500 pounds.  That means you'd need multiple trips even with the lightest moose. Something like weight scaling butcher times with also a cutoff point to where you don't need to bring it in at all?  Maybe anything that weighs less than 15 pounds can be butchered right there on the spot.

Most innawoods animals like deer for example can be butchered with just a short/long rope I'd say.  Almost all videos I see of hunters doing it [use ropes to hang it up and process it*GRAPHIC*](http://www.ski-epic.com/2009_deer_hunting_fort_peck_montana/p34s_3_bagging_deer_hanging_on_horse_corral.jpg).  You could say butchering it on a table with proper tools will yield more resource, and I agree.  But a lone cabin dweller should only need a rope and a knife with lots of time to spend cutting in order to get something.  Keep in mind that maybe rope isn't strong enough for all creatures!

I would say some tools could be added to make it easier for hunters.  In kitchens, you could spawn items like "[meat hook](http://www.collectiblesforeverythingnow.com/pics/222982763668_1.jpg)".  Could double as a weapon and also have the ability to hang animals up on a tree much faster or something.  30 minutes to hang up a corpse on a tree with rope vs 1 minute to skewer it and hang up, for example.  There are also much rarer [gambrels](https://www.discountramps.com/images/xxl/KS-GMB.jpg?v=100114425361-1) which could improve efficiency of cutting up meat and be a good mid-game item for hunters.  Maybe just double the grappling hook as a meat hook?

[Ability to cook a bone-in quartered chunk of meat](https://www.brokenarrowranch.com/Assets/Images/Wild%20Boar%20Leg%20Call%20Out.jpg) as like "cooked game meat" that seperates into portions depending on the size of the quarter.  A quartered moose might get you 6 game meats per quarter.  Not useful for more dedicated recipes but helps innawoods scenario early game.  Also heavy and not preservable but will give you a fair amount of nutrition.

TL;DR

There are many ways to improve the early game with the new butchering system.

- Tying 80-200 pound game up in trees and just cutting them up that way
- Allowing small animals (less than 10 lbs) to just be butchered just right in the field with no equipment
- adding a few items that allow innawoods to butcher without requiring advanced equipment.
-  Ability to cook game meat without the needs to butcher it at all, with the downside of not being able to readily preserve it or transport it.

**Comment by sfsworms** [comment]
Odieman: I think up to a point you might be mistaking field dressing for butchery. Most hunters will field dress the deer after roping it to a tree, but then take the carcass to a real butcher to do the preparation and harvest the meat. 

The current statu quo also allow you to do the "field butchery", which seems equivalent to what you're proposing.

**Comment by nexusmrsep** [comment]
@kevingranade I need some confirmation & decisions to avoid misunderstanding, so below I have summarized most commonly repeated feedback in form of some optional choices:

1. Should **simple (quick) butchery allow all yields but at low ratio** (1/4 of max, skill dependant)? Most common yields being meat, offal, skin, bones, fat, sinew, feather, wool and stomach, with some special swaps for triffids for example.

2. **Skinning** should be:
 - a part of quick butchery?
- a separate action?
- both?

If separate action, should it  allow butchering later? Should there be other interactions (allow/disallow, loose yield, etc.)  between skinning and other types of butchering?

**Fate of butchering rack and flat surfaces.**
(assuming that flat surface is to prevent freshly butchered meat from touching the ground, and a butchering rack is a necessary tool for easy access to all parts of the corpse, and disallowing it to touch the dirty ground during the process:)

3.  Should I:
- keep butchering rack and flat surface (freshly butchered meat should not touch the ground to stay edible) requirement, and implement alternative options (butchering rack can be replaced by a rope and a tree, flat surface can be emulated by )
- keep the butchering rack, kick out the flat surface requirement (implement butchering rack options)
- kick out both butchering rack and flat surface requirement, 
- make both optional, with flag BUTCHER_AID, implement a rule where you lose some yield (to what degree?) if you don't have those tools? (can't increase yields, there is just as much meat on the corpse)


If butchering rack and flat surface are to stay:

4. Should they be required for MS_SMALL and MS_TINY creatures?
5. Should they become obsolete after survival skill hits a threshold, and character is trained enough to do it without them on site? What level would it be?

6. Should sawing tools be always required for full butchering?

**Comment by Odieman1** [comment]
sfsworms, you're right.  For further processing of large game you'd need some sort of Tree stump table and a lot of time.

**Comment by FulcrumA** [comment]
In all honesty, I wouldn't mind it if butchering, rather than being several various option would be simply butchering and quick butchering, where all the tools, skills, location and other circumstances affect final result and so would be both with yield and a chance to retrieve CBMs. 

I understand if this opinion may be controversial but I don't personally need 4 different ways of preparing a corpse, merely results of preparing the corpse simulating yield under different circumstances that would be fair - and I'd rather develop the feature with this in mind rather than make it progressively more complex and micromanaging-requiring chore.

**Comment by Tsunder** [comment]
@Odieman1 good resources, thanks

I do want to note that for game balance purposes, butchering should not take very long, as the actaul gained resources of butchering are way less than IRL - a single chunk of meat from a small woodland animal, or a dozen from a larger animal, are generally much less than a real life moose or whatever would yield.

I think that one option is to make corpses take on a fractional butchering state, to be able to quickly get some resources from it (example, 1/4 butchered, or a leg or something), while the rest of it remains unbutchered for later more efficient usage.

**Comment by kevingranade** [comment]
> Should simple (quick) butchery allow all yields but at low ratio (1/4 of max, skill dependant)? Most common yields being meat, offal, skin, bones, fat, sinew, feather, wool and stomach, with some special swaps for triffids for example.

Short answer is yes, I think a number of people have made a solid case for simple butchering both having a solid realism rationale for recovering all types of resources (except CBMs) and having a solid game rationale for the same (innawoods characters with limited carrying capacity, bootstrapping innawoods crafting).

I think it's more productive to flip this around and say what parts *can* be spoiled:
Meat, yes
Offal, very yes (i.e. it's hard not to)
Skin, very yes (same)
Bones, no
Fat, I'm not sure, probably yes
Sinew, can probably fuck it up if you're in a hurry, but it's probably pretty easy to avoid given how tough they are.
Feathers, no, AFAIK its standard practice and rather fast to pluck a fowl before butchering.
Wool, yes, you're going to have to go out of your way to shear a sheep corpse properly.
Stomach, no idea, can a hunter help me out?
Triffid/fungal whatevers, not sure.
CBMs, very yes.

So what stands out to me there is bones and feathers.  Give full yields always.
Everything else falls into some degree of rushing is going to reduce yields.

Skinning is tricky, there are a number of valid-sounding options, but doing all of them is prohibitively complex in the short term.
Something that seems reasonably straightforward is having both quick butchering and full butchering yield skin, and everything else (except CBM removal, which destroys skin) ignores it.

Tool requirements for full butchering.
I think it's reasonable to lump the benefits of various tool requirements together and have those benefits be all-or-nothing.

AFAICT a "rope over a tree branch" is going to be substantially similar in effect to a rack.  The rack is better, but a few percent better, not 60% better.

It does seem reasonable to exempt small animals from the rack and table requirement.

Perhaps as skill level increases, fast butchering yields would increase and eventually catch up with full butchering (this would happen at a lower skill level the smaller the animal?), but full butchering should remain substantially faster.

Sawing tools for large animals does sound like a good requirement fie faster and more efficient butchering.

Fractional butchering:
This is a nice idea, but it sounds painful to implement and tune.  The current approach is already complex, and is a lot simpler than fractional butchering would be.

**Comment by nexusmrsep** [comment]
Ok, so let me enumerate things that I will do:

**Ad 1.**  Simple butchery will yield gamut of results, as described above. It will be capped at 1/4 maximum yields.

**Ad. 2.** Skinning will be available for quick and full butchery. I will reduce yields in quick butchery for creatures  of size MEDIUM and above, to simulate that you don't skin the whole animal, but just cut-off all accessible skin from the top. Rationale: Larger game would need the use of butchering rack / rope and tree to remove all it's skin at once, so that implies full butchery. That said those animals are big enough to cut some usable skin of a decent size while in a hurry - quick butchery. Small animals, like rabbits can be skinned whole either in quick or full butchery, so there will be no limit here.

**Ad 3.**  Butchering rack and requirement for flat surface to protect the meat stays for full butchering. Alternative methods will be implemented: Rope and a nearby tree will be a viable replacement of a butchering rack. Craftable fiber mat (made of bark, or dried plants) will be a viable replacement of a leather tarp / table / tourist table / tree stump.

**Ad 4.** TINY and SMALL animals will be fully-butcherable without a rack and flat surface.

** Ad 6.** Requirement for a sawing tool will be applied only to creatures of size MEDIUM and above.

-------------
**Ad 5.** This is actually a bit tricky, so this is what I think: 
Skill is presently used to determine final yield (success rate) of butchering for both quick/full butcheries. Implementing it to maximum yield limit for particular kind of butchery (quick) might lead to unexpected outcome, and finally will result in player being able to quick butcher a moose in a nick of time for max yields, while full butchering the same moose will take lot more time for the same amount of yield. This would be unaccountable. Then again if quick b. yields would stay at 1/4 of max yields (full b. yields) then doing full butchery would matter for yields sake, including time taken.
In conclusion with all the above options available we should either leave it alone as is, or disable the need fot the rack/rope&tree after lets say level 8 survival?

**Comment by vlad1492** [comment]
>Stomach, no idea, can a hunter help me out?

I only did a bit of hunting as a kid long ago, but yes you can ruin a stomach as a container through haste or sloppy knifework.   It would still be as edible, and still be useful for rennet. 

Fat - you will get less waste with careful trimming, but the bulk of it is readily available and doesn't need to be intact. It all goes in the rendering pot anyhow.

**Comment by vlad1492** [comment]
Another point- the meat explosions may have shifted from the sublime into the ridiculous.  Example:  Awl pike vs cougar, 27 pieces of meat scatter.  Multiply for a horde.

**Comment by nexusmrsep** [comment]
@Pixelmage I'm not against any idea here realy, but with many valid but conflicting ideas it's @kevingranade that calls the shots. Convince him. Though I believe you will be satisfied when quick butchering will simply more variety of goods.

**Comment by vlad1492** [comment]
Dissecting in the dark is weird.  Maybe make a light requirement for Full and Dissect?

**Comment by vlad1492** [comment]
Cosmetic issue perhaps: 

"You are not in the mood and the prospect of guts and blood on your hands convinces you to turn away."

I'm trying to  'Cut up all you can' on a pile of turnout gear while 'Sad'.
Individual pieces return a more appropriate 'prospect of work stops you before you begin'.

**Comment by Pixelmage** [comment]
Assuming everything continues on this track and doesn't shift into the racks being an optional boost (my preference), I'd like to toss a coin into your last note there:

> In conclusion with all the above options available we should either leave it alone as is, or disable the need for the rack/rope&tree after lets say level 8 survival?

I'd vote for eventually outgrowing the requirement at X survival. That way a character that does survive long enough to develop the skill earns a more streamlined process to go through the system. Perhaps make it so different sizes get freed at different tiers: TINY never needs a rack, SMALL stops needing it at 2, MEDIUM stops at 5 and LARGE at 8.

**Comment by kevingranade** [comment]
> Perhaps as skill level increases, fast butchering yields would increase and eventually catch up with full butchering (this would happen at a lower skill level the smaller the animal?), but full butchering should remain substantially faster.

If @nexusmrsep doesn't want to implement this suggestion (which is fine with me, it's a nice to have, not a requirement), we can capture it in a new issue.  This would mean you would eventually get full yields for all creatures, but theres always a benefit to having a full set of tools (butchering speed).

**Comment by vlad1492** [comment]
Finally a use for the UPS-powered electric carver!  
Right?

**Comment by narc0tiq** [comment]
> fast butchering yields would increase [...], but full butchering should remain substantially faster.

Just a note, it seems counter-intuitive to me that "quick" butchering would be slower than "full" butchering.

Overall, I think I'd be fine with merging the two back into one action (the distinction is meaningless to me as a player), where equipment and skill would provide better/faster results without being absolutely required. Of course, this leaves the problem of discoverability/incentive: how does a new player learn that they can get better butchery results if they had a rack and tarp? And why would they work toward acquiring the tools (skills being an eventual guarantee)?

**Comment by Pixelmage** [comment]
> Of course, this leaves the problem of discoverability/incentive: how does a new player learn that they can get better butchery results if they had a rack and tarp? And why would they work toward acquiring the tools (skills being an eventual guarantee)?

If the maximum yield threshold is high enough, the proportional results of butchering without tools at skill 0 would be terrible. When you cheese a moose kill on day one and get 1 piece of meat out of it because you only got 1/15th of the yield (I.E. LARGE animals requiring an effective skill of 12~15), you're bound to realize something is missing. Add to that good description to the tools in the crafting menu, so that the player looking for anything "butchery" related to craft will see items that explain "This makes butchering faster and more effective!" / "This allows you to perform dissections." should be enough. 

And if they power through, relying only the yield increases from natural skill, they still get to progress, even if at a slower rate.

To add to that - It might be a good idea to make it so all CBM-enabled corpses are either guaranteed or have a high chance to yield 1 Burnt-Out CBM upon butchering. As a flag to the player that they might want to dissect that type instead. Otherwise, how would you know what's worth the time and effort?

**Comment by narc0tiq** [comment]
> It might be a good idea to make it so all CBM-enabled corpses are either guaranteed or have a high chance to yield 1 Burnt-Out CBM upon butchering. As a flag to the player that they might want to dissect that type instead. Otherwise, how would you know what's worth the time and effort?

At the moment, I believe you're supposed to get a message in the log about destroying CBMs with your butchery, but yielding a non-functional CBM component as well would be nice.

I like the rest of your idea, too, I just want more explicit information for the player -- a message log line or maybe even a `query_yn` along the lines of "That [moose corpse] looks too big to butcher effectively without tools or experience. Try it anyway?" With the query or message no longer occurring once the player skills up.

---

Anecdotally, I had the darndest time trying to find the craftable butchery rack until I looked at the PR and found out it was actually a construction (oops!) -- and I knew it existed and was a thing I wanted. I'm not sure whether that needs to be more discoverable or if it's par for the course -- I can't remember how I did with other similar things in the past.

**Comment by kevingranade** [comment]
> Just a note, it seems counter-intuitive to me that "quick" butchering would be slower than "full" butchering.

How it works is shifting, so the name is becoming inaccurate.
Possibly "Field Butchering" (on the ground with a limited set of tools) vs "Butchering" (on furniture with a variety of tools).

> Anecdotally, I had the darndest time trying to find the craftable butchery rack until I looked at the PR and found out it was actually a construction (oops!) -- and I knew it existed and was a thing I wanted. I'm not sure whether that needs to be more discoverable or if it's par for the course -- I can't remember how I did with other similar things in the past.

It IS par for the course unfortunately.  Ideally we would unify the crafting and construction menus, possibly with just different tabs that display crafts vs constructions.  If nothing else it should be fairly easy to have a directory that just lists names and descriptions, but not necessarally the prerequisites and everything.

**Comment by Shinino** [comment]
A couple of things to poke along at this, from what I've done and been seeing:

1)  the change to deer/sheep and travois will directly help this, as a travois now has the ability to cart a medium sized creature innawoods to somewhere (presuming a campsite) for better butchering, removing the need for non-large creatures from being butchered there and then.

2)  A PR is up to allow vines to be used instead of ropes (which affects travois as well).  I'd go a step further, and say that we should have a way (as said earlier) to skin an animal, and then manipulate that skin, giving another way to rig travois with 'hide strips' as a rope/vine replacement.

**Comment by kevingranade** [comment]
>
>
>
>    1.
>
>    A PR is up to allow vines to be used instead of ropes (which affects
>    travois as well). I'd go a step further, and say that we should have a way
>    (as said earlier) to skin an animal, and then manipulate that skin, giving
>    another way to rig travois with 'hide strips' as a rope/vine replacement.
>
>
Is skin tough enough to drag around the carcass immediately after
butchering? I was under the impression that it needed significant treatment
to get it ready for that kind of thing.

>

**Comment by kevingranade** [comment]
AFAIK the skinning action was the last requirement from this issue.
If you want to make sure other aspects of this issue are preserved, please open a new issue.
